src/FormMigration/Steps/MailchimpSettings.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Give\FormMigration\Steps;
+
+use Give\FormMigration\Contracts\FormMigrationStep;
+use Give\Framework\Blocks\BlockModel;
+
+/**
+ * @unreleased
+ */
+class MailchimpSettings extends FormMigrationStep
+{
+    /**
+     * @unreleased
+     */
+    public function process(): void
+    {
+        $prevFormId = $this->formV2->id;
+
+        $isFormEnabled = $this->getFormMetaValue($prevFormId, '_give_mailchimp_enable') === 'true';
+        $isFormDisabled = $this->getFormMetaValue($prevFormId, '_give_mailchimp_disable') === 'true';
+        $isGloballyEnabled = give_get_option( 'give_mailchimp_show_checkout_signup') === 'on';
+
+        if ($isFormDisabled || (!$isGloballyEnabled && !$isFormEnabled)) {
+            return;
+        }
+
+        $block = BlockModel::make([
+            'name'       => 'givewp/mailchimp',
+            'attributes' => $this->getAttributes($prevFormId)
+        ]);
+
+        $this->fieldBlocks->insertAfter('givewp/email', $block);
+    }
+
+    /**
+     * @unreleased
+     */
+    private function getFormMetaValue(int $prevFormId, string $metaKey)
+    {
+        $meta = give()->form_meta->get_meta($prevFormId, $metaKey, true);
+
+        return $meta === '' ? null : $meta;
+    }
+
+    /**
+     * @unreleased
+     */
+    private function getAttributes($prevFormId): array
+    {
+        return   [
+            'label'            => $this->getFormMetaValue($prevFormId, '_give_mailchimp_custom_label') ??
```

Comment:
These would also be good additions to the `FormMetaDecorator`.

Reply:
We are just talking about the form meta values here correct? f8a30efecf86d0c0a15de8ecb42590bdd60ac82b